project/profiles/capacity4more/modules/c4m/user/c4m_user/c4m_user.module

```diff
@@ -677,15 +682,139 @@ function c4m_user_access_denied() {
 /**
  * Implements hook_menu_alter().
  *
- * Make the user/%user/contact page just a callback.
+ * Make the user/%user/contact page.
  */
 function c4m_user_menu_alter(&$items) {
   $items['user/login'] = array(
     'title' => 'Please log in to continue',
   );
   $items['user/%user/contact'] = array(
     'type' => MENU_CALLBACK,
+    'title' => 'User contact form',
+    'description' => 'User contact form.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('c4m_user_contact_form', 1),
+    'access arguments' => array('administer users'),
+  );
+}
+
+/**
+ * Create 'Contact User' form.
+ */
+function c4m_user_contact_form($form, &$form_state, $account) {
+  global $user;
+  $form_state['user'] = $account;
+
+  $form['c4m_user_from'] = array(
+    '#type' => 'textfield',
+    '#title' => t('From'),
+    '#description' => t("From address"),
+    '#size' => 30,
+    '#default_value' => $user->mail,
+    '#disabled' => TRUE,
+  );
+
+  $form['c4m_user_subject'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Subject'),
+    '#description' => t("Enter mail subject."),
+    '#size' => 10,
   );
+
+  $form['c4m_user_body'] = array(
+    '#title' => t('Body'),
+    '#description' => t('Enter mail body'),
+    '#type' => 'text_format',
+    '#format' => 'full_html',
+  );
+
+  $form['c4m_user_checkbox'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Send me a copy'),
+    '#default_value' => FALSE,
+  );
+
+  $form['c4m_user_submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Send'),
+  );
+
+  return $form;
+}
+
+/**
+ * Validate function for contact_form.
+ *
+ * Verifies that email subject and body are not empty.
+ */
+function c4m_user_contact_form_validate($form, $form_state) {
+  $subject = $form_state['values']['c4m_user_subject'];
+  $body = $form_state['values']['c4m_user_body']['value'];
+
+  if (empty($subject)) {
+    form_error($form['c4m_user_subject'], 'Subject field must have a value!');
+  }
+
+  if (empty($body)) {
+    form_error($form['c4m_user_body'], 'Body field must have a value!');
+  }
+}
+
+/**
+ * Submit function for contact_form.
+ *
+ * Generates required parameters and sends an email.
+ * If 'Send copy' checkbox is checked, sends copy mail to running user.
+ */
+function c4m_user_contact_form_submit($form, $form_state) {
+  global $user;
+
+  // Generate and send mail.
+  $account = $form_state['user'];
+  $subject = $form_state['values']['c4m_user_subject'];
+  $body = $form_state['values']['c4m_user_body']['value'];
+  $send_copy = $form_state['values']['c4m_user_checkbox'] == 1;
```

Comment:
@anvmn 
I believe that if the checkbox is unchecked then the value is `0` ?
In any case, `1` and `0` are also considered as _Boolean_, hence we don't need to make this comparison `... == 1`.

Remove this line and just use the value on the condition at the bottom.